### ISSUE TITLE
Flow misses type checks for constructor

### ISSUE THREAD

**Issue by Xeoncross** [yannickcr/eslint-plugin-react#1314]
In the following example I'm using Flow with React. I'm new to Flow and noticed that Flow doesn't catch the following string -> int conversion but React does. Am I doing something wrong? I assumed flow would track `props` and notice that `state.age` !== `props.age`.

```
// @flow

import React, { PropTypes } from 'react'

// In the following component we will see 'age' used as both a string and int

// These could instead be inline below (instead of pre-defined like this)
type FlowProps = {name: string, age: string}
type FlowState = {name: string, age: number}

class App extends React.Component {

  // For React (comment this out and everything 'works')
  static propTypes = {
    name: PropTypes.string,
    age: PropTypes.number
  }

  static defaultProps: FlowProps = {
    name: '',
    age: "foo"
  }

  // For Flow
  state: FlowState
  props: FlowState

  constructor(props: FlowState){
    super()
    this.state = {
      name: "bar",
      age: props.age // Flow misses this, but React.PropTypes catches it
    }
  }

  render () {
    return (
      <div>
        <Greet name={this.state.name} age={this.state.age}></Greet>
      </div>
    )
  }
}

const Greet = (props) => (<h1>{props.name} is {props.age}</h1>)

export default App;

```

Note: The solution that flow should find is that `constructor(props: FlowState)` should be `constructor(props: FlowProps)`.

**Comment by ljharb** [comment]
That sounds like a flow bug, not a bug with the react eslint plugin?

**Comment by Xeoncross** [comment]
That's what I thought https://github.com/facebook/flow/issues/4421

**Comment by ljharb** [comment]
This plugin lints that you're using React correctly; by that you're using Flow correctly - I don't think this plugin is the right place for it. Thanks for filing!
